models/classes/Tool/Factory/Lti1p3LaunchRequestFactory.php

```diff
@@ -0,0 +1,125 @@
+use oat\taoLti\models\classes\Tool\LtiLaunchCommandInterface;
+
+class Lti1p3LaunchRequestFactory extends ConfigurableService
+{
+    /** @var LtiLaunchRequestBuilder */
+    private $ltiLaunchRequestBuilder;
+
+    /** @var OidcLaunchRequestBuilder */
+    private $oidcLaunchRequestBuilder;
+
+    public function withLtiLaunchRequestBuilder(LtiLaunchRequestBuilder $ltiLaunchRequestBuilder): self
+    {
+        $this->ltiLaunchRequestBuilder = $ltiLaunchRequestBuilder;
+
+        return $this;
+    }
+
+    public function withOidcLaunchRequestBuilder(OidcLaunchRequestBuilder $oidcLaunchRequestBuilder): self
+    {
+        $this->oidcLaunchRequestBuilder = $oidcLaunchRequestBuilder;
+
+        return $this;
+    }
+
+    public function create(LtiLaunchCommandInterface $command): LaunchRequestInterface
+    {
+        $registration = $this->getRegistrationRepository()
+            ->find($command->getLtiProvider()->getId());
+
+        if ($command->isAnonymousLaunch()) {
+            return $this->getLtiLaunchRequestBuilder()->buildResourceLinkLtiLaunchRequest(
+                new ResourceLink($command->getResourceIdentifier(), $command->getLaunchUrl()),
+                $registration,
+                $registration->getDefaultDeploymentId(),
+                $command->getRoles(),
+                $command->getClaims()
+            );
+        }
+
+        if ($command->isOpenIdConnectLaunch()) {
+            return $this->getOidcLaunchRequestBuilder()->buildResourceLinkOidcLaunchRequest(
+                new ResourceLink($command->getResourceIdentifier(), $command->getLaunchUrl()),
+                $registration,
+                $command->getOpenIdLoginHint(),
+                $registration->getDefaultDeploymentId(),
+                $command->getRoles(),
+                $command->getClaims()
+            );
+        }
+
+        return $this->getLtiLaunchRequestBuilder()->buildUserResourceLinkLtiLaunchRequest(
+            new ResourceLink($command->getResourceIdentifier(), $command->getLaunchUrl()),
+            $registration,
+            $this->getUserIdentity($command),
+            $registration->getDefaultDeploymentId(),
+            $command->getRoles(),
+            $command->getClaims()
+        );
+    }
+
+    private function getLtiLaunchRequestBuilder(): LtiLaunchRequestBuilder
+    {
+        return $this->ltiLaunchRequestBuilder ?? new LtiLaunchRequestBuilder();
+    }
+
+    private function getOidcLaunchRequestBuilder(): OidcLaunchRequestBuilder
+    {
+        return $this->oidcLaunchRequestBuilder ?? new OidcLaunchRequestBuilder();
+    }
+
+    private function getRegistrationRepository(): RegistrationRepositoryInterface
+    {
+        return $this->getServiceLocator()->get(Lti1p3RegistrationRepository::class);
+    }
+
+    private function getUserIdentity(LtiLaunchCommandInterface $command): UserIdentityInterface
+    {
+        $user = $command->getUser();
```

Comment:
```suggestion
        if (!$user = $command->getUser()) {
            throw new \Exception('LtiLaunchCommand object does not contain user');
        }
```

Reply:
It is not necessary, cause I already check `$command->isAnonymousLaunch()` before, so it will never be null at this point.